### PR TITLE
[0-size Tensor Job2 No.66] Add 0-size Tensor support for paddle.nn.functional.softmax_with_cross_entropy

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

66 paddle.nn.functional.softmax_with_cross_entropy

soft_label 为False时，axis 所在列不能为0，其他列相同，所以 softmax, loss的numel 都为0，这里判断softmax 的numel为0
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/functional/softmax_with_cross_entropy_cn.html#softmax-with-cross-entropy
image

PaddleAPITest 测试通过，错误为numpy error
image

55 paddle.nn.functional.cross_entropy - part
soft_label 为True时，loss维度中axis所在列设置为1，返回填充0
其中，在python端，如果reduction为mean，返回填充需要修改为nan，因为需要调试性能问题，这部分没有修改，在其他PR再修改
PaddleAPITest 测试通过，错误为numpy error
image

### CODE REVIEW

**Comment by co63oc** [comment]
@DanielSun11 CI已完成需要review

**Review by DanielSun11 — APPROVED** [review]
LGTM
